plugins/quota/rest.go

```diff
@@ -2,29 +2,116 @@ package quota
 	"github.com/gluster/glusterd2/pkg/api"
 	"github.com/gluster/glusterd2/pkg/errors"
 	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
 )
 
-func commonValidator(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	volname := mux.Vars(r)["volname"]
-	v, err := volume.GetVolume(volname)
+//QuotadStart is the function to start the quota daemon
+func QuotadStart(c transaction.TxnCtx) error {
+	quotadDaemon, err := NewQuotad()
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
-	} else if v.State != volume.VolStarted {
-		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return err
+	}
+	// Create pidfile dir if not exists
+	err = os.MkdirAll(path.Dir(quotadDaemon.pidfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
 	}
+	// Create logFiledir dir
+	err = os.MkdirAll(path.Dir(quotadDaemon.logfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
+	}
+	err = daemon.Start(quotadDaemon, true)
+	if err == errors.ErrProcessAlreadyRunning {
+		c.Logger().WithError(err).Warn("Quota Daemon is already running.")
+		return nil
+	}
+	return err
 }
 
 func quotaEnableHandler(w http.ResponseWriter, r *http.Request) {
-	// implement the help logic and send response back as below
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volName := p["volname"]
+
 	ctx := r.Context()
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "todo: quota enable")
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate volume existence
+	vol, err := volume.GetVolume(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if quota is already enabled
+	if volume.IsQuotaEnabled(vol) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrProcessAlreadyRunning.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Enable quota
+	vol.Options[volume.VkeyFeaturesQuota] = "on"
+
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	if err := txn.Ctx.Set("volinfo", vol); err != nil {
+		logger.WithError(err).Error("failed to set volinfo in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	lock, unlock, err := transaction.CreateLockSteps(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "vol-option.UpdateVolinfo",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		{
+			DoFunc: "quota-enable.DaemonStart",
+			Nodes:  vol.Nodes(),
+		},
```

Comment:
Notify step required. Required only if Quota enabled on second volume and Quotad is already running. You can also add the step later while implementing force